### PR TITLE
refactor(common): extern template for motion_utils / remove tier4_autoware_utils.hpp / remove motion_utis.hpp

### PR DESCRIPTION
**Pull request by soblin**
## Description

https://github.com/autowarefoundation/autoware.universe/issues/4821

This PR
- removed tier4_autoware_utils.hpp/path_with_lane_id_geometry.hpp
- refactor motion_utils package
- removed motion_utils.hpp
- export template functions in motion_utils as `extern template`

## Tests performed

Psim works

## Effects on system behavior

Psim and AWSIM works

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by TakaHoribe** [comment]
@soblin Would you write the background of why the tier4_autoware_utils.cpp are removed in the readme for future developers?

**Review by satoshi-ota — APPROVED** [review]

**Review by tkimura4 — APPROVED** [review]

**Review by yukkysaito — APPROVED** [review]
LGTM

File: common/motion_utils/CMakeLists.txt [hunk 1]
```diff
@@ -7,13 +7,14 @@ autoware_package()
...
 ament_auto_add_library(motion_utils SHARED
-  src/motion_utils.cpp
   src/distance/distance.cpp
   src/marker/marker_helper.cpp
   src/marker/virtual_wall_marker_creator.cpp
   src/resample/resample.cpp
+  src/trajectory/trajectory.cpp
   src/trajectory/interpolation.cpp
   src/trajectory/path_with_lane_id.cpp
+  src/trajectory/tmp_conversion.cpp
```
Comment (yukkysaito) [inline common/motion_utils/CMakeLists.txt:17]: Why is this temporary?

Comment (soblin) [reply common/motion_utils/CMakeLists.txt:17]: Actually I do not know, `tmp_convesion.hpp` exists, so I followed the file name.

https://github.com/autowarefoundation/autoware.universe/blob/main/common/motion_utils/include/motion_utils/trajectory/tmp_conversion.hpp

Comment (yukkysaito) [reply common/motion_utils/CMakeLists.txt:17]: I see :+1:

**Review by miursh — APPROVED** [review]

**Comment by soblin** [comment]
@maxime-clem Can you approve this PR ?

**Review by maxime-clem — APPROVED** [review]

**Pull request merged**